Source/Orts.Simulation/Common/Scripting/BrakeController.cs

```diff
@@ -15,124 +15,210 @@
+        
         /// <summary>
         /// Current value of the brake controller
         /// </summary>
-        public Func<float> CurrentValue;
+        protected float CurrentValue() => Host.CurrentValue;
+
+        /// <summary>
+        /// Intermediate value of the brake controller
+        /// </summary>
+        protected float IntermediateValue() => Host.IntermediateValue;
+
         /// <summary>
         /// Minimum value of the brake controller
         /// </summary>
-        public Func<float> MinimumValue;
+        protected float MinimumValue() => Host.MinimumValue;
+
         /// <summary>
         /// Maximum value of the brake controller
         /// </summary>
-        public Func<float> MaximumValue;
+        protected float MaximumValue() => Host.MaximumValue;
+
         /// <summary>
         /// Step size of the brake controller
         /// </summary>
-        public Func<float> StepSize;
+        protected float StepSize() => Host.StepSize;
+
         /// <summary>
         /// State of the brake pressure (1 = increasing, -1 = decreasing)
         /// </summary>
-        public Func<float> UpdateValue;
+        protected float UpdateValue() => Host.UpdateValue;
+
         /// <summary>
         /// Gives the list of notches
         /// </summary>
-        public Func<List<MSTSNotch>> Notches;
+        protected List<MSTSNotch> Notches() => Host.Notches;
+
+        /// <summary>
+        /// Current notch of the brake controller
+        /// </summary>
+        protected int CurrentNotch() => Host.CurrentNotch;
 
         /// <summary>
         /// Sets the current value of the brake controller lever
         /// </summary>
-        public Action<float> SetCurrentValue;
+        protected void SetCurrentValue(float value)
+        {
+            Host.CurrentValue = value;
+        }
+
+        /// <summary>
+        /// Sets the intermediate value of the brake controller lever
+        /// </summary>
+        protected void SetIntermediateValue(float value)
+        {
+            Host.IntermediateValue = value;
+        }
+
         /// <summary>
         /// Sets the state of the brake pressure (1 = increasing, -1 = decreasing)
         /// </summary>
-        public Action<float> SetUpdateValue;
+        protected void SetUpdateValue(float value)
+        {
+            Host.UpdateValue = value;
+        }
+
+        /// <summary>
+        /// Sets the current notch of the brake controller lever
+        /// </summary>
+        protected void SetCurrentNotch(int notch)
+        {
+            Host.CurrentNotch = notch;
+        }
+
         /// <summary>
         /// Sets the dynamic brake intervention value
         /// </summary>
-        public Action<float> SetDynamicBrakeIntervention;
+        protected void SetDynamicBrakeIntervention(float value)
+        {
+            // TODO: Set dynamic brake intervention instead of controller position
```

Comment:
I fixed this TODO some days ago, so if you had the idea of changing this you don't have to.

Reply:
Well, if you have a different code available for this, I can integrate it in this PR too.

Reply:
The changes were introduced together with some changes to dynamic brake blending (see #732). Your PR seems harmless, so I think it's better to merge this first, and then I'll rebase #732.

This also conflicts with #713 (it adds a hook to command train brakes from cruise control), which is almost ready except for documentation.